geopandas/tests/util.py

```diff
@@ -45,12 +48,35 @@ def connect(dbname):
     return con
 
 
-def create_db(df):
+def create_sqlite(df, filename, geom_col="geom"):
+    """
+    Create a sqlite database with the nybb table.
+    """
+
+    con = sqlite3.connect(filename)
+    cur = con.cursor()
+    cur.execute("PRAGMA foreign_keys=OFF;")
+    cur.execute("BEGIN TRANSACTION;")
+    cur.execute("CREATE TABLE IF NOT EXISTS 'nybb' "
+                "( ogc_fid INTEGER PRIMARY KEY AUTOINCREMENT, "
+                "'{}' BLOB, 'borocode' INTEGER, ".format(geom_col) +
+                "'boroname' VARCHAR(32), 'shape_leng' FLOAT, "
+                "'shape_area' FLOAT);")
```

Comment:
I arrived it this using `ogr2ogr` to convert the nybb shapefile to a sqlite database, and then the `.dump` command in sqlite that provides the test to create the database.

Reply:
This is what ogr2ogr outputs as the command it uses to create the database?

(I was wondering a bit too what extent we are sure this is how a real spatialite file will look like, they mention on their website they are using a slight custom version of wkb)

Reply:
It is a two-stepper. I don't know what `ogr2ogr` is doing internally, but I did `ogr2ogr -f "sqlite" nybb.sqlite nybb.shp`. Then `spatialite nybb.sqlite`. Then in the `spatialite` console, `.dump nybb` to provide the text version of the command to create the database. Your concern about whether this is typical is reasonable. My reading on the custom format suggested that the fact that the column type of "BLOB" was what I should have expected. I originally intended to just leave the changes to use the `hex_encoded` kwarg with the expectation that that would be useful for future developments with other database types, but then after playing with `ogr2ogr2` outputs thought I had something that was actually working.